engine/access/rest/middleware/metrics.go

```diff
@@ -13,14 +13,17 @@ import (
 	"github.com/gorilla/mux"
 )
 
+// we have to use single rest collector for all metrics since it's not allowed to register same
+// collector multiple times.
+var restCollector = metrics.NewRestCollector(metricsProm.Config{Prefix: "access_rest_api"})
```

Comment:
there should only ever be a single REST server running on a node. Is `MetricsMiddleware` getting called multiple times in production?

Rather than using a global, how about we build the collector externally and pass it into `MetricsMiddleware()`, or pass in the prefix value?

Reply:
Wouldn't adding this to the MetricsMiddleware function be ok since it creates a specific middleware that is then being used on the server thus only initializing it once?

Reply:
I understand it will be a bit more refactoring, but I think we should align this with how metrics are used elsewhere and pass the collector in rather than create a special case.

what this may look like is creating the collector object in the AN's node builder, and passing it through into the middleware, similar to how it works with the logger. In tests that are not explicitly testing metrics, we can use the `NoopCollector` which is already setup with stubs for the `RestCollector`: https://github.com/onflow/flow-go/blob/16c25eaa5c3ae82a1c16fe3c6200184fdb325158/module/metrics/noop.go#L242-L247

A pattern we use in the network layer, is to embed sub-collectors in the top level struct and interface:
https://github.com/onflow/flow-go/blob/16c25eaa5c3ae82a1c16fe3c6200184fdb325158/module/metrics/network.go#L23-L29
that way the single top level collector can be passed through.

The RestCollector could be embedded in the AccessCollector [interface](https://github.com/onflow/flow-go/blob/16c25eaa5c3ae82a1c16fe3c6200184fdb325158/module/metrics.go#L572-L593) and [implementation](https://github.com/onflow/flow-go/blob/16c25eaa5c3ae82a1c16fe3c6200184fdb325158/module/metrics/access.go#L8-L16) in a similar way.